### PR TITLE
feat(pricing): add Qwen3.8 Flash and Max rates with full cache pricing

### PR DESCRIPTION
**Pull request by Alaric-L**
LiteLLM's together_ai/Qwen/Qwen3.8-Flash entry omits cache read/write costs and dashscope/qwen3.8-max omits cache-write, so cache-heavy dsh rows undercounted ~48x. Pin both models in curated-overrides.json (input/output/cache_read/cache_write per MTok) plus the canonical edge block mirrored verbatim into the four other edge files; cover exact/fuzzy/cased/dot lookup variants and a computeRowCost regression on a real cache-dominated row.

## Scope

- [ ] CLI (`src/`)
- [ ] Dashboard (`dashboard/`)
- [ ] macOS app (`TokenTrackerBar/`)
- [ ] Windows app (`TokenTrackerWin/`)
- [ ] Docs / CI / config

## Checklist

- [ ] `npm test` passes
- [ ] New user-facing strings go through `dashboard/src/content/copy.csv` (no hardcoded UI text)
- [ ] Commits follow conventional style (`feat:` / `fix:` / `refactor:` / `docs:` / `chore:` / `test:` / `ci:`)
- [ ] PR description explains *why*, not just *what*

---

<details>
<summary><strong>Risk layer addendum</strong> — expand if this PR touches any trigger below</summary>

### Risk layer triggers

- [ ] Public exposure / share links / unauthenticated access
- [ ] Auth / session / token handling
- [ ] Cross-endpoint invariants or shared logic
- [ ] External gateway / environment constraints

### Rules / invariants

-

### Boundary matrix (list at least 3)

-

### Public exposure checklist (if applicable)

- [ ] Public access rules defined (share token required, non-JWT handling, 401 behavior)
- [ ] Exposed fields explicitly listed and verified
- [ ] Avatar/image policy defined
- [ ] Regression tests cover invalid link and auth fallback

</details>

<details>
<summary><strong>Codex review context</strong> — fill when requesting <code>@codex</code> review</summary>

- **Delta since last Codex review:** (commits or summary)
- **Intended behavior / invariants:**
- **Edge cases covered:**
- **Tests run (command + result):**
- **Known gaps / out of scope:**

### Most likely regression surface

-

### Verification method (choose at least one)

-

### Uncovered scope

-

</details>

## Summary by CodeRabbit

* **New Features**
  * Added pricing support for Alibaba Qwen3.8 Flash and Qwen3.8 Max models.
  * Model identifiers are now matched consistently across capitalization, provider prefixes, date suffixes, and formatting variations.
  * Token usage costs include input, output, cache-read, and cache-write rates.

* **Bug Fixes**
  * Corrected cost calculations for Qwen3.8 model usage, including cache-heavy requests.

* **Tests**
  * Added coverage to verify pricing resolution and billing accuracy for both models.

### CODE REVIEW

File: test/pricing.test.js [hunk 1]
```diff
@@ -214,6 +214,38 @@ test("matcher: GLM-5.3 and GLM-5.3-Flash resolve to their own curated rates (not
...
+  // billed ~48x low. Curated must win even when LiteLLM has the model.
+  const litellm = {
+    "together_ai/Qwen/Qwen3.8-Flash": { input: 0.15, output: 0.47 },
+    "dashscope/qwen3.8-max": { input: 2, output: 6, cache_read: 0.25 },
+  };
+  const cases = [
+    ["qwen3.8-flash", 0.15, 0.47, 0.016, 0.2, "curated:exact"],
+    ["qwen3.8-max", 2, 6, 0.25, 2.5, "curated:exact"],
+    // cased / suffixed variants land on the right entry via fuzzy
+    ["Qwen3.8-Flash", 0.15, 0.47, 0.016, 0.2, "curated:fuzzy"],
```
[the remark this answers is not in the document]
Comment (Alaric-L) [reply test/pricing.test.js:231]: Verified against the actual matcher: buildDotRestoredModel only returns a non-empty dot-form when a \d-\d segment exists, so case-only variants like Qwen3.8-Flash have an empty dotForm and skip the exact-dot branch, resolving via curated:fuzzy. node test/pricing.test.js passes 57/57 on this checkout. Marking as false positive.

**Review by xiufengsun — CHANGES_REQUESTED** [review]
Reviewed exact head `59350f8e2d5e83528d0089598bbee52c056c3fb7`. All 57 pricing tests pass; the case-only lookup assertion is correct, so the earlier bot finding is not a blocker.

There is a separate pricing issue: the new `qwen3.8-max` entry applies `$0.25/M` to all cached reads while also adding explicit cache creation pricing. Alibaba's current Singapore/International table distinguishes **implicit cache = $0.25/M** from **explicit cache read = $0.17/M**, with explicit creation at $2.50/M. See the [official Qwen3.8 Max pricing table](https://www.alibabacloud.com/help/en/model-studio/qwen3-8-max).

Consequently, the cache-heavy Max fixture (`500,000` cached reads plus explicit cache creation) expects $0.389, while the corresponding explicit-cache case is $0.349. A global pin would overprice that workload, and the five edge copies repeat the same assumption. Merely changing the one shared rate to $0.17 would instead underprice implicit-cache usage.

Please establish the cache-mode semantics of the source counters with a counts-only fixture, preserve or distinguish that mode where pricing needs it, and cover both explicit and implicit cache cases across local and edge paths. Document the Singapore/International basis of the bare-model default and verify Flash's cache rates against the same first-party pricing surface. Keeping this open until cache modes are priced consistently.
